timeseries/src/autogluon/timeseries/trainer/abstract_trainer.py

```diff
@@ -878,15 +862,174 @@ def score(
-        data: TimeSeriesDataFrame,
         model: Union[str, AbstractTimeSeriesModel],
+        data: TimeSeriesDataFrame,
+        model_pred_dict: Dict[str, TimeSeriesDataFrame],
         known_covariates: Optional[TimeSeriesDataFrame] = None,
-        **kwargs,
     ) -> TimeSeriesDataFrame:
+        """Generate predictions using the given model.
+
+        This method assumes that model_pred_dict contains the predictions of all base models, if model is an ensemble.
+        """
         if isinstance(model, str):
             model = self.load_model(model)
-        data = self.get_inputs_to_model(model=model, X=data, known_covariates=known_covariates)
-        return model.predict(data, known_covariates=known_covariates, **kwargs)
+        data = self._get_inputs_to_model(model=model, data=data, model_pred_dict=model_pred_dict)
+        return model.predict(data, known_covariates=known_covariates)
+
+    def _get_inputs_to_model(
+        self,
+        model: str,
+        data: TimeSeriesDataFrame,
+        model_pred_dict: Dict[str, TimeSeriesDataFrame],
+    ) -> Union[TimeSeriesDataFrame, Dict[str, TimeSeriesDataFrame]]:
+        """Get the first argument that should be passed to model.predict.
+
+        This method assumes that model_pred_dict contains the predictions of all base models, if model is an ensemble.
+        """
+        model_set = self.get_minimum_model_set(model, include_self=False)
+        if model_set:
+            for m in model_set:
+                if m not in model_pred_dict:
+                    raise AssertionError(f"Prediction for base model {m} not found in model_pred_dict")
+            return {m: model_pred_dict[m] for m in model_set}
+        else:
+            return data
+
+    def get_model_pred_dict(
+        self,
+        model_names: List[str],
+        data: TimeSeriesDataFrame,
+        known_covariates: Optional[TimeSeriesDataFrame] = None,
+        record_pred_time: bool = False,
+        raise_exception_if_failed: bool = True,
+    ) -> Union[Dict[str, TimeSeriesDataFrame], Tuple[Dict[str, TimeSeriesDataFrame], Dict[str, float]]]:
+        """Return a dictionary with predictions of all models for the given dataset.
+
+        Parameters
+        ----------
+        model_names
+            Names of the model for which the predictions should be produced.
+        data
+            Time series data to forecast with.
+        known_covariates
+            Future values of the known covariates.
+        record_pred_time
+            If True, will additionally return the total prediction times for all models (including the prediction time
+            for base models). If False, will only return the model predictions.
+        raise_exception_if_failed
+            If True, the method will raise an exception if any model crashes during prediction.
+            If False, error will be logged and predictions for failed models will contain None.
+        """
+        if self.cache_predictions:
```

Comment:
Consider adding `use_cache` as an argument with default True, so that we can recompute even if we have the cache for the purposes of measuring predict times or as a sanity check.

Reply:
Nice idea, thanks!